tests/unit-tests/conversions.cpp

```diff
@@ -59,4 +51,107 @@ TEST_CASE("C++ Standard Conversions") {
       lexicon.double_type()
   );
   lexicon.make_if_then(condition, *lexicon.make_expr_stmt(then_expr));
+
+  // Bitcast                      (Pretend)
+  INFO("(bool*)ptr");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(ptr),
+    lexicon.get_pointer(lexicon.bool_type()),
+    lexicon.get_pointer(lexicon.bool_type())
+  );
+}
+
+TEST_CASE("Class Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  INFO("struct Base {};");
+  INFO("struct Derived : Base {};");
+  auto& base = *lexicon.make_class(*unit.global_region());
+  auto& derived = *lexicon.make_class(*unit.global_region());
+  derived.declare_base(base);
+
+  INFO("Base b;");
+  INFO("Derived d;");
+  auto& base_ptr = lexicon.get_pointer(base);
+  auto& derived_ptr = lexicon.get_pointer(derived);
+  auto& b = *unit.global_region()->declare_var(*lexicon.make_identifier("b"), base_ptr);
+  auto& d = *unit.global_region()->declare_var(*lexicon.make_identifier("d"), derived_ptr);
+
+  // Derived-to-base conversion
+  INFO("(Base*)b;");
+  lexicon.make_narrow(
+    *lexicon.make_id_expr(d),
+    base,
+    base_ptr
+  );
+
+  // Checked base-to-derived conversion
+  INFO("dynamic_cast<Derived*>(b);");
+  lexicon.make_widen(
+    *lexicon.make_id_expr(b),
+    derived,
+    derived_ptr
+  );
+
+  // Unchecked base-to-derived conversion
+  INFO("(Derived*)b;");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(b),
```

Comment:
`Pretend` is for different situations, not directly expressed in code.

Consider
``` 
const int i = 0;
int*p = &i;
```

The standard says that the type of the expression `*p` is `int`, not `const int` (as you have expected if instead of `int`, you had a class type `A`).  There is an automatic adjustment from `const int` to `int` -- that is performed only for non-class types.  That automatic adjustment is expected to be represented as a `Pretend [ source: Deref p, target: int ]`

Reply:
@GabrielDosReis I am a bit confused,  the code snippet above would not compile.

Reply:
@Xazax-hun I meant `p` to be declared with type `const int*`.